openmetrics/tests/test_openmetrics.py

```diff
@@ -0,0 +1,190 @@
+    assert aggregator.metrics_asserted_pct == 100.0
+
+def test_openmetrics_default_instance(aggregator, poll_mock):
+    """
+    Testing openmetrics with default instance
+    """
+
+    c = OpenMetricsCheck(CHECK_NAME, None, {}, [], default_instances={
+        'openmetrics': {
+            'prometheus_url': 'http://localhost:10249/metrics',
+            'namespace': 'openmetrics',
+            'metrics': [
+                {'metric1': 'renamed.metric1'},
+                'metric2'
+            ]
+        }},
+        default_namespace='openmetrics')
+    c.check({
+        'prometheus_url': 'http://custom:1337/metrics',
+    })
+    aggregator.assert_metric(CHECK_NAME + '.renamed.metric1', tags=['node:host1', 'flavor:test', 'matched_label:foobar'], metric_type=aggregator.GAUGE)
+    aggregator.assert_metric(CHECK_NAME + '.metric2', tags=['timestamp:123', 'node:host2', 'matched_label:foobar'], metric_type=aggregator.GAUGE)
+    assert aggregator.metrics_asserted_pct == 100.0
+
+def test_openmetrics_mixed_instance(aggregator, poll_mock):
+    """
+    Testing openmetrics with default instance
+    """
+
+    c = OpenMetricsCheck(CHECK_NAME, None, {}, [], default_instances={
+        'foobar': {
+            'prometheus_url': 'http://localhost:10249/metrics',
+            'namespace': 'foobar',
+            'metrics': [
+                'metric3',
+                'metric4'
+            ]
+        }, 'openmetrics': {
+            'prometheus_url': 'http://localhost:10249/metrics',
+            'namespace': 'openmetrics',
+            'metrics': [
+                'metric2'
+            ],
+            'label_joins': {
+                'metric2': {
+                    'label_to_match': 'matched_label',
+                    'labels_to_get': ['timestamp']
+                },
+            },
+            'tags': ['extra:bar']
+        }},
+        default_namespace='openmetrics')
+    # run the check twice for label joins
+    for _ in range(2):
+        c.check({
+            'prometheus_url': 'http://custom:1337/metrics',
+            'namespace': 'openmetrics',
+            'metrics': [
+                {'metric1': 'renamed.metric1'}
+            ],
+            'label_joins': {
+                'renamed.metric1': {
+                    'label_to_match': 'matched_label',
+                    'labels_to_get': ['flavor']
+                },
+            },
+            'label_to_hostname':'node',
+            'tags': ['extra:foo']
+        })
+
+    for m in aggregator._metrics.items():
+        print m
+
+    aggregator.assert_metric(CHECK_NAME + '.renamed.metric1', hostname="host1", tags=['node:host1', 'flavor:test', 'matched_label:foobar', 'timestamp:123', 'extra:foo'], metric_type=aggregator.GAUGE)
+    aggregator.assert_metric(CHECK_NAME + '.metric2', hostname="host2", tags=['timestamp:123', 'node:host2', 'matched_label:foobar', 'extra:foo'], metric_type=aggregator.GAUGE)
```

Comment:
These are reeeeally long lines. Why do we need the `setup.cfg` file, and why is `max-line-length` set to 700 ?